src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php

```diff
@@ -139,6 +144,8 @@ function (array $language) {
                         'id' => $language['iso_code'],
                         'value' => sprintf('%s - %s', $language['iso_code'], $language['name']), ];
                 }, $this->get('prestashop.adapter.legacy.context')->getLanguages()),
+            'multistoreForcedContextInfoTip' => $this->getMultistoreAllShopsContextForcedMessage(),
```

Comment:
These two lines are repeated in multiple actions inside the controller, why not to create a protected method that returns an array to be merged with other view variables?

Reply:
@arouiadib what will be the benefits of such change?

Instead of having multiple times
```
'multistoreForcedContextInfoTip' => $this->getMultistoreAllShopsContextForcedMessage(),
```

you will have multiple times
```
return $this->render('@PrestaShop/Admin/Configure/ShopParameters/OrderStates/index.html.twig', $data);
$data = array_merge($someDataJustForThisController), $this->getCommonVariables());
```

What is better in this? 🤔  the only gain I see is it avoids repeating the array key `multistoreForcedContextInfoTip`. Is this change worth the additionnal changes?

These usecases "should I make this a reusable function" usecase are tricky 😉  if you ask 3 developers "should we make this a reusable method" (add an abstraction) you might have 3 different answers. Personally when I dont see an obvious gain in readability and maintainability, I prefer not to add an abstraction. I'm afraid of wrong overengineered abstractions, you can find many articles about it
- https://alexkondov.com/the-cost-of-wrong-abstractions/
- https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction
- https://www.reddit.com/r/programming/comments/5txp5t/duplication_is_far_cheaper_than_the_wrong/

Reply:
That was exactly the aim of my question: learning again from @matks. Thanks a lot.